src/hooks/Rule/bridge/useInfluxdbLineProtocol.ts

```diff
@@ -0,0 +1,99 @@
+interface KeyValueItem {
+  key: string
+  value: string
+}
+
+interface ParseResult {
+  measurement: string
+  tagArr: Array<KeyValueItem>
+  fieldArr: Array<KeyValueItem>
+  timestamp: number | undefined
+}
+
+export default (): {
+  parseLine: (line: string) => ParseResult | undefined
+  convertArrToMap: (arr: Array<KeyValueItem>) => Record<string, string>
+} => {
+  /* 
+    # Syntax
+    <measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>]
+
+    # Example
+    myMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000
+  */
+
+  /*
+    can use there sample to confirm func
+    const SQL1 = 'weather,location=us-midwest,closeSea=true temperature=82,humidity=71 1465839830100400200'
+    const SQL2 = `my\ Measurement,tag\ Key1=tag\ Value1,tag\ Key2=tag\ Value2,tagKey=🍭 fieldKey="\"string\" within a string",fieldKey2="Launch 🚀"`
+   */
+
+  const measurementReg = /[^,]+/
+  // TODO:Consider the case of spaces or commas inside string value
+  const tagItemReg = /(?<tagKey>[^\s,]+)=(?<tagValue>[^\s,]+)/
+  const tagPartReg = new RegExp(`(${tagItemReg.source},?)+`)
+  const fieldItemReg = /(?<fieldKey>[^\s,]+)((?<!\/)=)(?<fieldValue>[^\s,]+)/
+  const fieldsPartReg = new RegExp(`(${fieldItemReg.source},?)+`)
+  const timestampPart = /\d+/
+
+  const protocolReg = new RegExp(
+    `^(?<measurement>${measurementReg.source})(,(?<tags>${tagPartReg.source}))?(\\s(?<fields>${fieldsPartReg.source}))(\\s(?<timestamp>${timestampPart.source}))?$`,
+  )
```

Comment:
I remember there is a place to manage Reg expression constants: https://github.com/emqx/emqx-dashboard-web-new/blob/master/src/common/constants.ts#L49

But there is nothing wrong with writing it here.

Reply:
because ip reg can be used in many places but these regs just can be used in this place